src/nb_enum.cpp

```diff
@@ -150,6 +173,21 @@ bool enum_from_python(const std::type_info *tp, PyObject *o, int64_t *out, uint8
     if (!t)
         return false;
 
+    if ((t->flags & (uint32_t) enum_flags::flag_enum) != 0 && Py_TYPE(o) == t->type_py) {
```

Comment:
What situation would cause the first part of the condition to be `true`, while the second part is `false`?